### PR TITLE
✨Introduce response stream writer util

### PR DESCRIPTION
**Pull request by calebcordry**
Takes a given response and writes it to a given writer. To be used for ad streaming.

Part of #27189

### CODE REVIEW

File: extensions/amp-a4a/0.1/streaming-utils.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Decodes readable stream from response and writes to given writeable stream.
+ * This function should be replaced with transform stream when well supported.
+ * @param {!Window} win
+ * @param {!Response} response
+ * @param {!WritableStreamDefaultWriter} writer
+ * @return {!Promise}
+ */
+export function streamResponseToWriter(win, response, writer) {
```
Comment (lannka) [inline extensions/amp-a4a/0.1/streaming-utils.js:25]: where is this used? could you add it to this PR?

Comment (calebcordry) [reply extensions/amp-a4a/0.1/streaming-utils.js:25]: sure, was trying to keep it simple but can make a larger pr.

Comment (lannka) [reply extensions/amp-a4a/0.1/streaming-utils.js:25]: was wondering how this method is going to be used. you can still leave TODOs in the caller to make the PR not too large.

File: extensions/amp-a4a/0.1/test/test-streaming-utils.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+const chunk2 = `
+   <style amp4ads-boilerplate>body{visibility:hidden}</style>
+   <script async src="https://cdn.ampproject.org/amp4ads-v0.js"></script>
+ </head>
+ <body>Hello, AMP4ADS world.</body>
+ </html>
+`;
+
+describes.fakeWin('streamResponseToWriter', {}, (env) => {
```
Comment (powerivq) [inline extensions/amp-a4a/0.1/test/test-streaming-utils.js:36]: How do we want to handle an interrupted stream?

File: extensions/amp-a4a/0.1/streaming-utils.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-a4a/0.1/streaming-utils.js]: Added the calling context.

**Comment by calebcordry** [comment]
Moved util to `src/utils`

File: extensions/amp-a4a/0.1/test/test-streaming-utils.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-a4a/0.1/test/test-streaming-utils.js]: Do you mean if the browser cancels the response? I'm not sure, but if that is the same as calling `.cancel` on the stream, the writer will be closed and if there is no content we will collapse the ad. If there is a body I think we will render a partial ad. If there is no `cancel` called I think we will just wait until we receive a body, though maybe we should time out. Not sure I am answering your question, let me know if you want to chat about this.

**Comment by calebcordry** [comment]
@dvoytenko friendly ping :) I need your approval now that the new files live in `src/`

**Pull request merged**